app/src/main/java/ai/brokk/gui/components/TokenUsageBar.java

```diff
@@ -287,7 +287,7 @@ public void setFragments(List<ContextFragment> fragments) {
         // Dispose prior subscriptions bound to this component before rebinding
         ComputedSubscription.disposeAll(this);
 
-        this.fragments = List.copyOf(fragments);
+        this.fragments = fragments.stream().filter(ContextFragment::isValid).toList();
```

Comment:
So just to double check - we still exclude invalid fragments from the context (e.g., invalid summaries). If so, then we're all good here on that end.

Other issue is to consider that this calls `@Blocking`, so we need to check if this is an EDT method

Reply:
we exclude invalid fragments from the *token usage bar*, but we show them as chips